caikit/runtime/model_management/loaded_model.py

```diff
@@ -110,9 +123,27 @@ def wait(self):
             try:
                 self._model = self._caikit_model_future.result()
             except CaikitRuntimeException:
-                if self._fail_callback:
-                    self._fail_callback()
-                raise
+                if self._retries:
+                    self._retries -= 1
+                    log.debug(
+                        "Failed to load %s from %s. Retrying with %d retries left",
+                        self.id,
+                        self.path,
+                        self._retries,
+                    )
+                    self._caikit_model_future = self._caikit_model_future_factory()
+                    # Try waiting again with a fresh load future. This may open
+                    # a recursive retry.
+                    try:
+                        self.wait()
+                    except CaikitRuntimeException:
+                        if self._fail_callback:
+                            self._fail_callback()
```

Comment:
won't this potentially call the fail callback `n` times? One for each retry failure in the case of many retry failures

Reply:
That seems... quite likely 🤦. Covid coding at its finest!